suave/backends/redis_store_backend.go

```diff
@@ -0,0 +1,116 @@
+package backends
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	suave "github.com/ethereum/go-ethereum/suave/core"
+	"github.com/go-redis/redis/v8"
+)
+
+var ffStoreTTL = 24 * time.Hour
+
+type RedisStoreBackend struct {
+	ctx      context.Context
+	cancel   context.CancelFunc
+	redisUri string
+	client   *redis.Client
+}
+
+func NewRedisStoreBackend(redisUri string) *RedisStoreBackend {
+	return &RedisStoreBackend{
+		cancel:   nil,
+		redisUri: redisUri,
+	}
+}
+
+func (r *RedisStoreBackend) Start() error {
+	if r.cancel != nil {
+		r.cancel()
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	r.cancel = cancel
+	r.ctx = ctx
+
+	client, err := connectRedis(r.redisUri)
+	if err != nil {
+		return err
+	}
+	r.client = client
+
+	return nil
+}
+
+func (r *RedisStoreBackend) Stop() error {
+	if r.cancel == nil || r.client == nil {
+		panic("Stop() called before Start()")
```

Comment:
hmmm perhaps better return an error instead of `panic`? i think panic shouldn't be used in live code if at all possible (and if used the function should be prefixed with `Must`)

Reply:
in this case, you could just silently return too